igtdetect/glossharvester.py

```diff
@@ -1,7 +1,17 @@
+import re
+import os
 
 def get_utterance(row: str):
     return row.split(':')[1].strip('\n') if ':' in row else 'NA'
 
+def get_utterance_and_prefix(row: str):
+    utterance = get_utterance(row)
+    prefix = detect_prefix(utterance)
+    if prefix:
+        for pf in prefix:
+            utterance = utterance.replace(pf, '')
```

Comment:
I think this might be too strong. If your prefix is 'a.' then `Jan collected the data.` becomes `Jan collected the dat`

Reply:
I would make a slice using the length of the prefix `utterance[len(pf):]`